### ISSUE TITLE
Reload does not discard changes but discards colorization

### ISSUE THREAD

**Issue by rwhirn** [TablePlus/TablePlus-Windows#110]
1. Which driver are you using and version of it (Ex: PostgreSQL 10.0):
MySQL 5.7.17

2. Which TablePlus build number are you using (the number on the welcome screen, Ex: build 81):
90

3. The steps to reproduce this issue:
change 2 fields in each of 3 records 
hit ctrl-r to reload
changes appear gone
hit ctrl-s
changes are still going to be made
![image](https://user-images.githubusercontent.com/25158630/66168228-4dbbff00-e602-11e9-98b0-8b30a65b3831.png)

**Comment by rwhirn** [comment]
colorization reappeared AFTER I changed another field

**Comment by huyphams** [comment]
Thanks, I've updated the reload UX. TablePlus now will show a confirm popup if there are pending changes. We can choose discard and reload or commit the changes before reloading or close the popup to cancel the reload.

**Comment by rwhirn** [comment]
Maybe it also has something to do with clicking "Apply". I know I had done that before the repload in the example above. Now, it just happened again. I made changes, THEN I clicked a different apply. Now all my colorization is gone, but ctrl-s still shows I have changes to confirm.

**Comment by huyphams** [comment]
Please try this patch and let me know how's it going: https://www.dropbox.com/s/qw87iunfkchtb85/TablePlusUpdate.exe?dl=0
Thanks @rwhirn

**Comment by rwhirn** [comment]
That worked for the "Reload" issue, but it doesn't help the "Apply" issue.

**Comment by huyphams** [comment]
I don't quite understand `THEN I clicked a different apply` mean, could you explain it? Thanks

**Comment by rwhirn** [comment]
A different filter. I had filtered out 8 records and made changes to a couple of them. Then I thought of a better filter which might pull a few more records that needed the same change. So I did that filter and clicked apply on that. So I got the same records I had before plus some new ones. But now the changes I made were not showing - the records looked like they had reverted back to their previous values, but if I clicked ctrl-s all the changes were still there in the confirm popup. In my case, I was glad because I didn't mean to lose them. But really, if you do a new filter it should probably discard your changes. It could do the same new popup you did for reload asking if you want to commit your changes or discard them. Tho I kind of hate those popups confirming I want to lose my changes, cause it slows you down. But every once in a while it saves you.

**Comment by huyphams** [comment]
Actually, I want to port this behavior from macOS, but I've had time yet, I will improve it in the next update.
![2019-10-03 22 45 56](https://user-images.githubusercontent.com/3654622/66175355-a2c23a00-e62f-11e9-80db-b536d0c9b13d.gif)

**Comment by huyphams** [comment]
I think this build should work out the `apply` issue:
https://www.dropbox.com/s/qw87iunfkchtb85/TablePlusUpdate.exe?dl=0

**Comment by huyphams** [comment]
Please let me know if you're still seeing the bug.

**Comment by rwhirn** [comment]
Well... your fix works. The colorization is there after "apply" a filter but only if the record or records that were changed appear in the new query. I just applied a new filter that left off the changed records. Now I have no indicator that there are unsaved changes to the table. And I could do many, many more filtered queries and not know. Will it remember those changes forever? 

I don't think this is reasonable. If you do a new query I would expect it to start "fresh". So I think it would be better to discard changes or use the same popup confirmation you made to ask if you want to discard changes.
